script.grab.fanart/service.py

```diff
@@ -0,0 +1,366 @@
+import json
+import random
+from kodi_six import xbmc, xbmcgui
+from future.moves._thread import start_new_thread
+import resources.lib.utils as utils
+
+class GrabFanartService:
+    current_mode = 'random'
+    
+    monitor = None #xbmc monitor object
+    WINDOW = None #object representing the home window
+    xbmc_tv = None #array for tv shows
+    xbmc_movies = None #array for movie files
+    xbmc_music = None #array for music artist
+
+    tv_index = 0
+    movie_index = 0
+    music_index = 0
+    
+    def __init__(self):
+        utils.log("Grab Fanart Service Started")
+        
+        #setup the window and file list
+        self.WINDOW = xbmcgui.Window(10000)
+        self.xbmc_tv = list()
+        self.xbmc_movies = list()
+        self.xbmc_music = list()
+
+        #let XBMC know the script is not ready yet
+        self.WINDOW.setProperty('script.grab.fanart.Ready',"")
+
+        #start populating the arrays right away - don't use threads here
+        if(utils.getSetting('mode') == '' or utils.getSetting('mode') == 'random'):
+            self.grabRandom()
+        else:
+            self.grabRecent()
+        
+        #start the monitor
+        self.monitor = UpdateMonitor(after_scan=self.updateMedia)
+        
+    def run(self):
+        
+        #keep this thread alive
+        while(True):
```

Comment:
while not xbmc.Monitor().abortrequested()

Reply:
I like waitForAbort here since each image item needs to be active for X seconds and then refreshed. See line 145.

Reply:
Disregard